File: DateDifference/DateDifference1.java
package DateDifference;
import java.time.*;
import java.util.*;

public class DateDifference1{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        while (true) {
            LocalDate date1 = askDate(scanner, "first");
            LocalDate date2 = askDate(scanner, "second");

            if (date1.isAfter(date2)) {
                LocalDate temp = date1;
                date1 = date2;
                date2 = temp;
            }

            Period difference = Period.between(date1, date2);
            StringBuilder result = new StringBuilder("Elapsed time: ");
            boolean hasDifference = false;

            if (difference.getYears() != 0) {
                result.append(difference.getYears()).append(difference.getYears() == 1 ? " year" : " years");
                hasDifference = true;
            }
            if (difference.getMonths() != 0) {
                if (hasDifference) result.append(", ");
                result.append(difference.getMonths()).append(difference.getMonths() == 1 ? " month" : " months");
                hasDifference = true;
            }
            if (difference.getDays() != 0) {
                if (hasDifference) result.append(", ");
                result.append(difference.getDays()).append(difference.getDays() == 1 ? " day" : " days");
                hasDifference = true;
            }

            if (!hasDifference) {
                result.append("No difference. Both dates are the same.");
            }

            System.out.println(result);
            System.out.println("________________________________________________________");
        }
    }

    private static LocalDate askDate(Scanner scanner, String order) {
        while (true) {
            try {
                System.out.print("Enter the " + order + " date (DD MM YYYY): ");
                int day = scanner.nextInt();
                int month = scanner.nextInt();
                int year = scanner.nextInt();
                return LocalDate.of(year, month, day);
            } catch (InputMismatchException e) {
                System.out.println("Error: Please enter valid numbers.");
                scanner.nextLine(); // Clear the buffer
            } catch (DateTimeException e) {
                System.out.println("Error: Invalid date. Please try again.");
                scanner.nextLine(); // Clear the buffer
            }
        }
    }
}

File: DateDifference/DateDifference1.py
from datetime import date

def ask_date(order):
    while True:
        user_input = input(f"Enter the {order} date (DD MM YYYY or 'exit' to quit): ").strip()
        if user_input.lower() == "exit":
            return None
        try:
            parts = user_input.split()
            if len(parts) != 3:
                raise ValueError("Please enter exactly three numbers: DD MM YYYY.")
            day, month, year = map(int, parts)
            return date(year, month, day)
        except ValueError as e:
            print(f"Error: {e} Please try again.")

def main():
    while True:
        date1 = ask_date("first")
        if date1 is None:
            print("Exiting program.")
            break

        date2 = ask_date("second")
        if date2 is None:
            print("Exiting program.")
            break

        if date1 > date2:
            date1, date2 = date2, date1  # Swap dates if the first is after the second

        years = (date2.year - date1.year) - ((date2.month, date2.day) < (date1.month, date1.day))
        months = ((date2.year - date1.year) * 12 + date2.month - date1.month) - (date2.day < date1.day)
        remaining_days = (date2 - date(date2.year, date2.month, date1.day if date2.day >= date1.day else 1)).days

        result_parts = []
        if years:
            result_parts.append(f"{years} year{'s' if years != 1 else ''}")
        if months % 12:
            result_parts.append(f"{months % 12} month{'s' if months % 12 != 1 else ''}")
        if remaining_days:
            result_parts.append(f"{remaining_days} day{'s' if remaining_days != 1 else ''}")

        result = "Elapsed time: " + ", ".join(result_parts) if result_parts else "No difference. Both dates are the same."
        print(result)
        print("_" * 56)

if __name__ == "__main__":
    main()
